### PR TITLE
Changed write_csv and read_csv to operate on real CSV files

### PR DESCRIPTION
**Pull request by micheles**
The trick is to add the right number of commas in the pre-header row and to quote the last field. As a bonus, now github can display properly the outputs of the engine.
This is part of https://github.com/gem/oq-engine/issues/5045

### CODE REVIEW

File: openquake/qa_tests_data/classical/case_1/expected/hazard_curve-PGA.csv [hunk 1]
```diff
@@ -1,3 +1,3 @@
-# generated_by='OpenQuake engine 3.5.0-gitd83c42cda6', start_date='2019-05-12T07:12:05', checksum=141718627, kind='rlz-000', investigation_time=1.0, imt='PGA'
+#,,,,,"generated_by='OpenQuake engine 3.7.0-git5e91b20b88', start_date='2019-09-05T06:13:25', checksum=141718627, kind='rlz-000', investigation_time=1.0, imt='PGA'"
```
Comment (raoanirudh) [inline openquake/qa_tests_data/classical/case_1/expected/hazard_curve-PGA.csv]: Why all the commas at the beginning?

**Comment by mmpagani** [comment]
Michele, all, can we have a meeting (perhaps next week) to consolidate the
output formats?
Grazie
Marco

**Comment by micheles** [comment]
Absolutely @mmpagani , we are going to have a meeting with the risk guys and the hazard team should be involved too, particularly with outputs like the event based ruptures.

File: openquake/qa_tests_data/classical/case_1/expected/hazard_curve-PGA.csv (hunk 1 above)
Comment (micheles) [reply openquake/qa_tests_data/classical/case_1/expected/hazard_curve-PGA.csv]: I originally put the commas at the beginning but then it looks bad on GitHub :-(

**Comment by raoanirudh** [comment]
There are some decent recommendations about managing metadata for csv files and groups of related tabular files from the CSV working group: https://github.com/w3c/csvw

Particularly:
[CSV on the Web: A Primer](http://www.w3.org/TR/2016/NOTE-tabular-data-primer-20160225/)
[Model for Tabular Data and Metadata on the Web](https://www.w3.org/TR/2015/REC-tabular-data-model-20151217/)
[Metadata Vocabulary for Tabular Data](http://www.w3.org/TR/2015/REC-tabular-metadata-20151217/)

**Comment by micheles** [comment]
This is very interesting, @raoanirudh, but for me having a separate metadata file is a no-go. We already too many files. The pre-header looks better IMO.

**Pull request merged**